src/Compiler/Checking/CheckDeclarations.fs

```diff
@@ -5171,11 +5172,17 @@ let CheckOneImplFile
         env,
         rootSigOpt: ModuleOrNamespaceType option,
         synImplFile) =
-
-    let (ParsedImplFileInput (_, isScript, qualNameOfFile, scopedPragmas, _, implFileFrags, isLastCompiland, _)) = synImplFile
+    
+    let (ParsedImplFileInput (fileName, isScript, qualNameOfFile, scopedPragmas, _, implFileFrags, isLastCompiland, _)) = synImplFile
     let infoReader = InfoReader(g, amap)
 
     cancellable {
+        use _ = 
+            Activity.start "CheckDeclarations.CheckOneImplFile" 
+                [|
+                    "fileName", fileName
+                    "qualifiedNameOfFile", qualNameOfFile.Text
+                |]
```

Comment:
It may probably also make sense to record the project file name or some other id of it, as a single file can be analysed multiple times in a context of several target frameworks, and it could be useful to distinguish them.

Reply:
Nodes in the trace have parents and children. One can always see which project task this file-checking task belonged to, see eg: 
<img width="1571" alt="image" src="https://user-images.githubusercontent.com/2478401/196284228-a022af7f-d6b4-4e2d-b02d-7dda901425e7.png">

This doesn't currently include TargetFramework information - that could be added on the project-level nodes, alongside project path/name.

Reply:
Each 'project + target framework' considered a different FCS project, so it covers everything needed.

Reply:
Except that we don't currently tag TargetFramework, so we'd have two project nodes with the same project paths and no TF information.

Reply:
> EDIT: I'm wrong - we tag project output path, which includes TF, so you're right 🙂Everything is included, if a bit hidden.

This is what we do in Rider, others probably rely on projects having different project stamps and outputs. It probably makes sense to tag something of these, if such projects would conflate otherwise?

We add the tfm to the project name because a few years ago it seemed like the best option for us for some reason which I can't remember anymore 🙂 Probably it was some more expensive equality checking or cache clearing when projects had the same path. VS used some other approach too prior to introducing stamps IIRC.